packages/trader/src/Modules/Trading/Components/Elements/Multiplier/__tests__/cancel-deal-mobile.spec.tsx

```diff
@@ -0,0 +1,143 @@
+
+        const info_icon = screen.getByTestId(/dt_popover_wrapper/i);
+        userEvent.hover(info_icon);
+
+        expect(screen.getByText(deal_cancellation)).toBeInTheDocument();
+        expect(screen.getByText(/Cancel your trade/i)).toBeInTheDocument();
+    });
+    it('should render <RadioGroup /> if has_cancellation === true and cancellation_range_list contains proper info', () => {
+        const new_mocked_props = { ...default_mocked_props, has_cancellation: true };
+        default_mock_store.modules = {
+            trade: {
+                cancellation_range_list: [{ value: '60m', text: 'mocked text' }],
+            },
+        };
+        const mock_root_store = mockStore(default_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        expect(screen.getByText('mocked text')).toBeInTheDocument();
+    });
+    it('should call onToggleCancellation if user clicked on Deal cancellation checkbox and onToggleDealCancel returned true', () => {
+        const mock_root_store = mockStore(default_mock_store);
+        render(mockCancelDeal(mock_root_store, default_mocked_props));
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
+
+        expect(mockOnToggleCancellation).toBeCalled();
+    });
+    it('should show <DealCancellationWarningDialog /> if (has_take_profit || has_stop_loss) && should_show_cancellation_warning is equal to true and user clicked on Deal cancellation checkbox', () => {
+        const new_mocked_props = { ...default_mocked_props, has_take_profit: true };
+        const new_mock_store = { ...default_mock_store, ui: { should_show_cancellation_warning: true } };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        expect(screen.queryByText(dialog_text)).not.toBeInTheDocument();
+        expect(screen.queryByText(check_box_text)).not.toBeInTheDocument();
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
+
+        expect(screen.getByText(dialog_text)).toBeInTheDocument();
+        expect(screen.getByText(check_box_text)).toBeInTheDocument();
+    });
+    it('should call toggleCancellationWarning function if cancellation warning checkbox inside of <DealCancellationWarningDialog /> was clicked', () => {
+        const new_mocked_props = { ...default_mocked_props, has_take_profit: true };
+        const new_mock_store = {
+            ...default_mock_store,
+            ui: { should_show_cancellation_warning: true, toggleCancellationWarning: jest.fn() },
+        };
+        const mock_root_store = mockStore(new_mock_store);
+        render(mockCancelDeal(mock_root_store, new_mocked_props));
+
+        const checkbox_with_cancellation = screen.getByText(deal_cancellation);
+        userEvent.click(checkbox_with_cancellation);
```

Comment:
ditto

Reply:
By default `<DealCancellationWarningDialog />` is closed. User needs to click on 'Deal cancellation' checkbox in order to open it and only then to click on the checkbox inside of it. Of course, I can add 'expect', that after first click `<DealCancellationWarningDialog />` was rendered, but actually it was tested in the case above. So I saw no point in testing it additionally in each test case if a have a separate case for it.
Can add if you think it is reasonable

Reply:
@kate-deriv Do you mean that the fact that `const checkbox_with_show = screen.getByText(check_box_text);` is found (doesn't throw an error) means that `userEvent.click(checkbox_with_cancellation);` lead to the desired (expected) result?
It would be great to make it explicit:
```
const checkbox_with_cancellation = screen.getByText(deal_cancellation);
const checkbox_with_show = screen.queryByText(check_box_text);
expect(checkbox_with_show).not.toBeInTheDocument();
userEvent.click(checkbox_with_cancellation);
expect(checkbox_with_show).toBeInTheDocument();
userEvent.click(checkbox_with_show);
```